independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateInstance.java

```diff
@@ -178,6 +179,48 @@ default TemplateInstance onRendered(Runnable action) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Sets the {@code locale} attribute that can be used to localize parts of the template, i.e. to specify the locale for all
+     * message bundle expressions in the template.
+     *
+     * @param locale a language tag
+     * @return self
+     */
+    default TemplateInstance setLocale(String locale) {
+        return setAttribute(LOCALE, Locale.forLanguageTag(locale));
+    }
+
+    /**
+     * Sets the {@code locale} attribute that can be used to localize parts of the template, i.e. to specify the locale for all
+     * message bundle expressions in the template.
+     *
+     * @param locale a {@link Locale} instance
+     * @return self
+     */
+    default TemplateInstance setLocale(Locale locale) {
+        return setAttribute(LOCALE, locale);
+    }
+
+    /**
+     * Sets the variant attribute that can be used to select a specific variant of the template.
+     *
+     * @param variant the variant
+     * @return self
+     */
+    default TemplateInstance setVariant(Variant variant) {
```

Comment:
```suggestion
    default TemplateInstance setSelectedVariant(Variant variant) {
```

I'd rather be more specific here...

Reply:
I thought about using it, but the `selected` seems redundant when using the code completion IMHO. I'll change it

Reply:
Hm, yes, maybe it make sense if we remove the `setVariants()` method...

Reply:
I removed the `setVariants` method but kept the `setVariant`. WDYT?

Reply:
:+1: makes sense.